Description: Fix escape() usage
Origin: upstream, https://fedorahosted.org/augeas/changeset/021ea39f8e400225e2d01b4c62eb9d56404f2ecd/
Bug: https://fedorahosted.org/augeas/ticket/242
Forwarded: not-needed
Reviewed-By: Igor Pashev <pashev.igor@gmail.com>
Last-Update: <2012-04-17>

--- augeas-0.10.0.orig/src/regexp.c
+++ augeas-0.10.0/src/regexp.c
@@ -50,7 +50,7 @@ char *regexp_escape(const struct regexp
     ret = fa_restrict_alphabet(r->pattern->str, strlen(r->pattern->str),
                                &nre, &nre_len, 2, 1);
     if (ret == 0) {
-        pat = escape(nre, nre_len);
+        pat = escape(nre, nre_len, RX_ESCAPES);
         free(nre);
     }
 #endif
